Back/Reciplastk.Gateway/Services/CustomerService.cs

```diff
@@ -0,0 +1,116 @@
+﻿using Azure;
+using Reciplastk.Gateway.DataAccess;
+using Reciplastk.Gateway.Models;
+using System;
+using System.Linq;
+
+namespace Reciplastk.Gateway.Services
+{
+    public class CustomerService
+    {
+        private readonly ReciplastkContext db;
+        
+        public CustomerService(ReciplastkContext db)
+        {
+            this.db = db;
+
+        }
+        public HttpResponseModel GetAll()
+        {
+            var customer = db.Customers.Where(x=>x.Isactive == true).ToList();
+            var response = new HttpResponseModel();
+            response.WasSuccessful = true;
+            response.Data = customer;
+            return response;
+        }
+        public HttpResponseModel GetById(int customerid)
+        {
+            var customer = GetById((int?)customerid);
+            var response = new HttpResponseModel();
+            if (customer != null)
+            {
+                response.WasSuccessful = true;
+                response.Data = customer;
+            }
+            else
+            {
+                response.WasSuccessful = false;
+                response.StatusMessage = "The customer was not found";
+            }
+            return response;
+        }
+        private Customer GetById(int? customerId)
+        {
+            var customer = db.Customers.FirstOrDefault(x => x.Customerid == customerId && x.Isactive == true);
+            return customer;
+        }
+        private Customer GetByNit(String customerNit)
+        {
+            var customer = db.Customers.FirstOrDefault(x => x.Nit == customerNit && x.Isactive == true);
+            return customer;
+        }
+        public HttpResponseModel Create(CustomerViewModel customerViewModel)
+        {
+            var response = new HttpResponseModel();
+            var customer = GetByNit(customerViewModel.nit);
```

Comment:
Make sure the customer does not exist with the same nit. 
```suggestion
            var customer = GetByNit(customerViewModel.nit);
            if(customer != null) // Customer exists with the same id
            {
             return new HttpResponseModel() {wasSuccessfull = false; message = "Ya existed un cliente con este mismo NIT"};
            }
```